### PR TITLE
KAFKA-7795 Produce a WARN log when duplicate config keys are found

### PR DESCRIPTION
**Pull request by mumrah**
This change simply adds some formatted error logging if duplicate keys are found when parsing a Kafka config file. Since we use Properties (which blindly accepts duplicates), the validation must happen when the Properties instance is loaded from a file.

No behavior changes here, just extra logging.

If I edit the broker config like:

```diff
diff --git a/config/server.properties b/config/server.properties
index 898d8ebc28..593651fd33 100644
--- a/config/server.properties
+++ b/config/server.properties
@@ -133,4 +133,8 @@ zookeeper.connection.timeout.ms=6000
 # The default value for this is 3 seconds.
 # We override this to 0 here as it makes for a better out-of-the-box experience for development and testing.
 # However, in production environments the default value of 3 seconds is more suitable as this will help to avoid unnecessary, and potentially expensive, rebalances during application startup.
-group.initial.rebalance.delay.ms=0
\ No newline at end of file
+group.initial.rebalance.delay.ms=0
+
+zookeeper.connection.timeout.ms=7000
+zookeeper.connection.timeout.ms=8000
```

It will produce the following error output at the top of the log:
```
[2019-01-10 10:54:28,679] WARN Duplicate config entries detected:
	Overwriting zookeeper.connection.timeout.ms = 6000 with 8000
	Overwriting zookeeper.connection.timeout.ms = 7000 with 8000
 (org.apache.kafka.common.utils.Utils)
```

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by ijuma** [comment]
Thanks for the PR. This should be a `WARN` IMO. `ERROR` should be reserved for cases where something is definitely broken.

File: clients/src/main/java/org/apache/kafka/common/utils/Utils.java [hunk 1]
```diff
@@ -551,20 +551,49 @@ public static String formatBytes(long bytes) {
...
             try (InputStream propStream = Files.newInputStream(Paths.get(filename))) {
-                props.load(propStream);
+                duplicateCheckingProps.load(propStream);
             }
         } else {
             System.out.println("Did not load any properties since the property file is not specified");
         }
 
+        if (!overwritten.isEmpty()) {
+            StringBuilder msg = new StringBuilder("Duplicate config entries detected!");
```
Comment (ijuma) [inline clients/src/main/java/org/apache/kafka/common/utils/Utils.java]: Nit: we don't normally use exclamation marks in Kafka log messages.

File: clients/src/main/java/org/apache/kafka/common/utils/Utils.java [hunk 2]
```diff
@@ -551,20 +551,49 @@ public static String formatBytes(long bytes) {
...
+            overwritten.forEach((key, values) -> {
+                values.forEach(value -> {
+                    msg.append("\n\tIgnoring ")
+                            .append(key)
+                            .append(" = ")
+                            .append(value);
```
Comment (ijuma) [inline clients/src/main/java/org/apache/kafka/common/utils/Utils.java]: Would we want to include the config value that we are not ignoring?

**Review by ijuma — COMMENTED** [review]
The approach we're using here seems a bit hacky since `loadProps` is a general method. It also means that we have no tests for the change. It seems like it would be cleaner for the method to return a `Multimap` and we could then have a utility method to print keys with more than one value. What do you think?

File: clients/src/main/java/org/apache/kafka/common/utils/Utils.java (hunk 2 above)
Comment (mumrah) [reply clients/src/main/java/org/apache/kafka/common/utils/Utils.java]: Yea good idea. We are printing all the configs later in the log, but seeing it here in context of the duplicate values would be useful.

**Comment by mumrah** [comment]
Definitely agree on it being a hack. I took your suggestion and separated the logic for parsing from validating. Where all do you think it would be useful to load Properties in this way (logging the warning)? There are quite a few usages of `Utils#loadProps`. For now I've only switched `kafka.Kafka` to use the new methods.

**Comment by mumrah** [comment]
retest this please

**Comment by mumrah** [comment]
retest this please

File: clients/src/test/java/org/apache/kafka/common/utils/UtilsTest.java [hunk 3]
```diff
@@ -466,4 +470,35 @@ public void testRecursiveDelete() throws IOException {
...
+            Map<Object, List<Object>> propMap = Utils.loadPropsMap(absPath);
+            propMap.values().forEach(values -> assertEquals(values.size(), 1));
+
+            Properties props = Utils.stripDuplicateProps(propMap);
+            assertEquals(props.getProperty("prop1"), "value1");
+            assertEquals(props.getProperty("prop2"), "value2");
+            assertEquals(props.getProperty("prop3"), "value3");
+        }
+
+        {
```
Comment (cmccabe) [inline clients/src/test/java/org/apache/kafka/common/utils/UtilsTest.java]: Can we make these two separate `@Test` clauses?

**Comment by cmccabe** [comment]
LGTM after comment is addressed.

**Comment by mumrah** [comment]
retest this please

**Pull request closed without merging**